core/embed/rust/src/ui/display/toif.rs

```diff
@@ -72,41 +67,46 @@ pub fn icon(icon: &Icon, center: Point, fg_color: Color, bg_color: Color) {
 /// Holding toif data and allowing it to draw itself.
 #[derive(PartialEq, Eq, Clone, Copy)]
 pub struct Toif {
-    pub data: &'static [u8],
-    pub name: &'static str, // useful for debugging purposes.
-    pub size: Offset,
-    pub format: ToifFormat,
+    data: &'static [u8],
 }
 
 impl Toif {
-    pub fn new(named_toif: NamedToif) -> Self {
-        let info = unwrap!(toif_info(named_toif.0));
-        Self {
-            data: named_toif.0[TOIF_HEADER_LENGTH..].as_ref(),
-            name: named_toif.1,
-            size: info.0,
-            format: info.1,
-        }
+    pub fn new(data: &'static [u8]) -> Self {
+        Self { data }
+    }
+
+    pub fn format(&self) -> ToifFormat {
+        let info = unwrap!(toif_info(self.data));
```

Comment:
unconvinced about those -- we're calling `toif_info` all over the place uselessly.

it seems slightly better to just implement `display_toif_info` in Rust here:

1. `new()` will assert the TOIF header
2. `format()` will match on the format character
3. `width()` and `height()` will just parse out the respective bytes
4. `size()` will combine `width()` + `height()`

all of those _look like_ they could be `const fn`s too, with the possible exception of `new()`?

Reply:
Good idea, implemented in [9ecb7b0](https://github.com/trezor/trezor-firmware/pull/2792/commits/9ecb7b0e4bcd9437f6b9b763ff4a3238cad03e3d)